prophyc/parsers/isar.py

```diff
@@ -72,14 +72,26 @@ def make_typedef(elem):
     elif "primitiveType" in elem.attrib and elem.get("name") not in primitive_types.values():
         return model.Typedef(elem.get("name"), primitive_types[elem.get("primitiveType")])
 
-def make_enum_member(elem):
+def make_enum_member(elem, warn):
     value = elem.get('value')
-    value = value if value != "-1" else "0xFFFFFFFF"
+    try:
+        int_value = int(value, 0)
+        if int_value < 0:
+            value = "0x{:X}".format(0x100000000 + int_value)
+    except ValueError as e:
+        if warn:
+            warn(str(e))
     return model.EnumMember(elem.get("name"), expand_operators(value))
 
-def make_enum(elem):
+def make_enum(elem, warn):
     if len(elem):
-        return model.Enum(elem.get("name"), [make_enum_member(member) for member in elem])
+        values = set()
```

Comment:
Consider refactoring duplicate-checking logic to separate function.
It could be private module-scope function (_check...) or function defined inside make_enum function.